src/Core/Form/IdentifiableObject/DataHandler/SpecificPriceFormDataHandler.php

```diff
@@ -0,0 +1,145 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command\AddProductSpecificPriceCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command\EditProductSpecificPriceCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\ValueObject\InitialPrice;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\ValueObject\SpecificPriceId;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime;
+
+class SpecificPriceFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(
+        CommandBusInterface $commandBus
+    ) {
+        $this->commandBus = $commandBus;
+    }
+
+    public function create(array $data): int
+    {
+        $fixedPrice = isset($data['leave_initial_price']) && $data['leave_initial_price'] ?
+            InitialPrice::INITIAL_PRICE_VALUE :
+            (string) $data['fixed_price']
+        ;
+
+        $command = new AddProductSpecificPriceCommand(
+            (int) $data['product_id'],
+            $data['reduction']['type'],
+            (string) $data['reduction']['value'],
+            (bool) $data['include_tax'],
+            $fixedPrice,
+            (int) $data['from_quantity']
+        );
+
+        $this->fillRelations($command, $data);
+
+        /** @var SpecificPriceId $specificPriceId */
+        $specificPriceId = $this->commandBus->handle($command);
+
+        return $specificPriceId->getValue();
+    }
+
+    public function update($id, array $data): void
+    {
+        $command = new EditProductSpecificPriceCommand((int) $id);
+        $this->fillRelations($command, $data);
+
+        if (isset($data['leave_initial_price']) && $data['leave_initial_price']) {
+            $command->setFixedPrice(InitialPrice::INITIAL_PRICE_VALUE);
+        } elseif (isset($data['fixed_price'])) {
+            $command->setFixedPrice((string) $data['fixed_price']);
+        }
+
+        if (isset($data['from_quantity'])) {
+            $command->setFromQuantity((int) $data['from_quantity']);
+        }
+        if (isset($data['date_range']) && array_key_exists('from', $data['date_range'])) {
```

Comment:
```suggestion
        if (!empty($data['date_range']['from'])) {
```

wdyt @jolelievre ?

Reply:
No it won't work The problem I was fixing was when the date input is returned empty, the data returned is 

```php
$data [
    'date_range' => [
        'from' => null,
    ]
]
```

We must enter in the `if` condition when the field is present in the array key `from` even when it's `null`, if not the value is not change, and since we do partial update the from field can never be emptied

I thought `isset` was enough because unlike `empty` I thought it didn't check the value, but turns out when value is `null` `isset` is falsy So the only way to detect this correctly is to check the key explicitly

I could modify the `DateRangeType` to set it an empty default data with `''` then the isset would work fine, but I prefer not to modify this form type which is used in other places And besides with this modif the handler will be more resilient regardless of which empty data the input may use in the future (if we decide to change it)